test/jdk/java/lang/reflect/code/CoreBinaryOpsTest.java

```diff
@@ -63,19 +63,19 @@
 public class CoreBinaryOpsTest {
 
     @CodeReflection
-    @SupportedTypes(types = {int.class, long.class, boolean.class})
+    @SupportedTypes(types = {int.class, long.class, byte.class, short.class, char.class, boolean.class})
```

Comment:
Alas we cannot have constant array values, hence the duplication. But, we can use enums:
```
    enum Types {
        INTEGRAL_BOOLEAN(
                new Class<?>[]{int.class, long.class, byte.class, short.class, char.class, boolean.class}),
        INTEGRAL_FLOATING_POINT(
                new Class<?>[]{int.class, long.class, byte.class, short.class, char.class, float.class, double.class}
        );

        final Class<?>[] types;

        PrimitiveTypeList(Class<?>[] types) {
            this.types = types;
        }
    }

    @Retention(RetentionPolicy.RUNTIME)
    @Target(ElementType.METHOD)
    @interface SupportedTypes {
        PrimitiveTypeList[] value();
    }

...

    @CodeReflection
    @SupportedTypes(Types.INTEGRAL_BOOLEAN)
    static int add(int left, int right) {
        return left + right;
    }
```
Up to you.

Reply:
Good idea, done!